src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -248,12 +256,12 @@ case class AppDefinition(
     for (pms <- portMappings) yield pms.map(_.servicePort.toInt)
 
   def hostPorts: Seq[Int] =
-    containerHostPorts.getOrElse(ports.map(_.toInt))
+    containerHostPorts.getOrElse(portDefinitions.map(_.port))
```

Comment:
:+1: renamed AppDefinition.port to AppDefinition.portNumbers and made it a lazy val.